Shopify/hydrogen
# Allow specifying an environment to use variable from

### Blocked

Currently based off of the work done here:

- https://github.com/Shopify/hydrogen/pull/861

### WHY are these changes introduced?

A follow up to both https://github.com/Shopify/hydrogen/pull/861 and https://github.com/Shopify/hydrogen/pull/809 that allows developers to specify which environment to use.

When these commands were first added/updated we defaulted to always pulling the `Preview` environment variables but there are cases where developers may want to use a different environment for testing purposes.

### WHAT is this pull request doing?

We recently shipped the ability to specify a `branchName` argument in the Admin API for the `environmentVariables` field. This PR hooks into that field and adds a new flag to `commonFlags` to facilitate it.

### HOW to test your changes?

1. Pull the branch and navigate to the `cli` directory
1. Run `npm run build`
1. Navigate to the demo store directory `cd ../../templates/demo-store`
1. Run `h2 env pull --branch <BRANCH>` (note: if `h2` is unavailable run `npx shopify hydrogen shortcut` first) to pull your specified environment variables down
   - You should be prompted to provide a shop
   - You should then be prompted to authenticate with that shop
   - You should be prompted that you need to link to a storefront
   - **Note:** replace `<BRANCH>` with the branch associated to an environment on your linked shop (e.g. `main` for most Production environments)
1. Confirm that the variables pulled are from your specified branch's environment

#### Checklist
